src/components/tools/table-tool/table-tool.tsx

```diff
@@ -73,33 +97,107 @@ export default class TableToolComponent extends BaseComponent<IProps, IState> {
-        defaultPrecision={1}
-        itemFlags={itemFlags}
-        readOnly={readOnly}
-        onGridReady={this.handleGridReady}
-        onSetAttributeName={this.handleSetAttributeName}
-        onAddCanonicalCases={this.handleAddCanonicalCases}
-        onSetCanonicalCaseValues={this.handleSetCanonicalCaseValues}
-        onRemoveCases={this.handleRemoveCases}
-      />
+      <div className="table-tool" ref={this.domRef}
+          tabIndex={this.props.tabIndex} onKeyDown={this.handleKeyDown} >
+        <DataTableComponent
+          dataSet={this.state.dataSet}
+          changeCount={this.state.syncedChanges}
+          autoSizeColumns={this.getContent().isImported}
+          indexValueGetter={this.indexValueGetter}
+          attrValueFormatter={this.attrValueFormatter}
+          defaultPrecision={1}
+          itemFlags={itemFlags}
+          readOnly={readOnly}
+          onGridReady={this.handleGridReady}
+          onSetAttributeName={this.handleSetAttributeName}
+          onAddCanonicalCases={this.handleAddCanonicalCases}
+          onSetCanonicalCaseValues={this.handleSetCanonicalCaseValues}
+          onRemoveCases={this.handleRemoveCases}
+        />
+      </div>
     );
   }
 
   private getContent() {
     return this.props.model.content as TableContentModelType;
   }
 
+  private initializeHotKeys() {
+    this.hotKeys.register({
+      "cmd-c": this.handleCopy,
+      "cmd-v": this.handlePaste
+    });
+  }
+
   private handleGridReady = (gridReadyParams: GridReadyEvent) => {
     this.gridApi = gridReadyParams.api || undefined;
     this.gridColumnApi = gridReadyParams.columnApi || undefined;
   }
 
+  private handleMouseDown: EventListener = (e: MouseEvent) => {
+    const target: HTMLElement = e.target as HTMLElement;
+    const targetClasses = target && target.className;
+    // don't mess with focus if this looks like something ag-grid has handled
+    if (typeof targetClasses !== "string") return;
+    if (targetClasses.includes("ag-cell") || targetClasses.includes("ag-header-cell")) {
+      return;
+    }
+
+    // table tile should have keyboard focus -- requires tabIndex
+    this.domRef.current && this.domRef.current.focus();
+
+    // clicking on table background clears selection
+    this.gridApi && this.gridApi.deselectAll();
+    this.gridApi && this.gridApi.refreshCells();
+  }
+
+  private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
+    this.hotKeys.dispatch(e);
+  }
+
+  private handleCopy = () => {
+    const { dataSet } = this.state;
+    if (this.gridApi && dataSet) {
+      const rowIds = this.gridApi.getSelectedNodes().map(row => row.id);
+      if (rowIds && rowIds.length) {
+        const { clipboard } = this.stores;
+        const clipData = {
+                attrs: dataSet.attributes.map(attr => ({ id: attr.id, name: attr.name })),
+                cases: dataSet.getCanonicalCases(rowIds)
+              };
+        clipboard.clear();
```

Comment:
Looks to me like the clipboard is not clearing properly for some reason - if I copy one row and then another, I end up pasting both.

Reply:
🤦‍♂️ Thx, I'll take a look.